source/interprocedural_analyses/taint/taintConfiguration.ml

```diff
@@ -223,9 +223,35 @@ let parse source_jsons =
     | `String "parametric" -> AnnotationParser.Parametric
     | unexpected -> failwith (Format.sprintf "Unexpected kind %s" (Json.Util.to_string unexpected))
   in
+  let unknown_keys = String.Hash_set.create () in
+  let check_keys ~required_keys ~valid_keys ~current_keys ~section =
+    let valid_keys_hash_set = String.Hash_set.of_list valid_keys in
+    let current_keys_hash_set = String.Hash_set.of_list current_keys in
+    let check_required_key_present key =
+      Hash_set.mem current_keys_hash_set key
+      |> function
+      | false ->
+          failwith (Format.sprintf "Required key %s is not found in section `%s`" key section)
+      | true -> ()
+    in
+    let check_key_is_valid key =
+      Hash_set.mem valid_keys_hash_set key
+      |> function
+      | false ->
+          Hash_set.add unknown_keys (Format.sprintf "Unknown key `%s` in section `%s`" key section)
```

Comment:
Could we just log the error here? I'm not a fan of this `unknown_keys` variable. It's easy to forget.